### DISCUSSION TITLE
请问如何才能正确部署到 cloudflare page 呢？

### DISCUSSION THREAD

**Discussion by Tikas** [vuepress-theme-hope/vuepress-theme-hope#1244] [Q&A] [answered]
最近想部署到 cloudflare page，然而，选择 vuepress 总是部署失败（虽然能正常部署，其实页面是 404）

下面是提示的一些警告

<html>
<body>

00:54:10.469 | [3/4] Linking dependencies...
-- | --
00:54:10.479 | warning " > vuepress-theme-hope@1.21.0" has unmet peer dependency "typescript@^4.4.4".
00:54:10.480 | warning "vuepress-theme-hope > vuepress-plugin-typescript@0.3.1" has unmet peer dependency "typescript@*".
00:54:10.481 | warning "vuepress-theme-hope > @docsearch/js > @docsearch/react@3.0.0-alpha.42" has unmet peer dependency "react@>= 16.8.0 < 18.0.0".
00:54:10.481 | warning "vuepress-theme-hope > @docsearch/js > @docsearch/react@3.0.0-alpha.42" has unmet peer dependency "react-dom@>= 16.8.0 < 18.0.0".
00:54:10.482 | warning "vuepress-theme-hope > @vuepress/plugin-blog > vuepress-plugin-sitemap@2.3.1" has incorrect peer dependency "chalk@^2.0.0".
00:54:10.483 | warning "vuepress-theme-hope > @vuepress/plugin-blog > vuepress-plugin-sitemap@2.3.1" has unmet peer dependency "commander@^2.0.0".
00:54:10.483 | warning "vuepress-theme-hope > @vuepress/plugin-blog > vuepress-plugin-sitemap@2.3.1" has unmet peer dependency "esm@^3.0.0".
00:54:10.483 | warning "vuepress-theme-hope > vuepress-plugin-typescript > ts-loader@7.0.5" has unmet peer dependency "typescript@*".
00:54:10.484 | warning "vuepress-theme-hope > @docsearch/js > @docsearch/react > @algolia/autocomplete-preset-algolia@1.5.0" has unmet peer dependency "@algolia/client-search@^4.9.1".
00:54:10.486 | warning "vuepress-theme-hope > @mr-hope/vuepress-plugin-comment > @mr-hope/vuepress-types > @types/markdown-it-anchor > markdown-it-anchor@8.4.1" has unmet peer dependency "markdown-it@*".

</body>
</html>

执行时，明显少了好多 app

<html>
<body>

00:54:21.630 | Verify run directory
-- | --
00:54:21.630 | Executing user command: vuepress build
00:54:22.725 | wait Extracting site metadata...
00:54:22.730 | tip Apply theme @vuepress/theme-default ...
00:54:22.753 | tip Apply plugin container (i.e. "vuepress-plugin-container") ...
00:54:22.754 | tip Apply plugin @vuepress/register-components (i.e. "@vuepress/plugin-register-components") ...
00:54:22.755 | tip Apply plugin @vuepress/active-header-links (i.e. "@vuepress/plugin-active-header-links") ...
00:54:22.755 | tip Apply plugin @vuepress/search (i.e. "@vuepress/plugin-search") ...
00:54:22.756 | tip Apply plugin @vuepress/nprogress (i.e. "@vuepress/plugin-nprogress") ...
00:54:23.777 | ℹ Compiling Client
00:54:23.797 | ℹ Compiling Server
00:54:33.168 | ✔ Server: Compiled successfully in 9.37s
00:54:38.055 | ✔ Client: Compiled successfully in 14.28s
00:54:38.113 | wait Rendering static HTML...

</body>
</html>

由于我对 vue 并不熟悉，查阅了网上很多网站，都没能解决这个问题，所以才来这里寻找解决办法。

**Comment by Mister-Hope** [comment]
1. 这跟本项目没太大关系
2. 你没有给一个项目的仓库地址，你指望我就这几行日志说点啥
3. 我有足够的理由怀疑你配置错了构建命令。

**Reply by Tikas** [reply]
哇，大佬竟然这么快回复，由于第一次使用 vuepree，也是第一次使用构建命令，确实很多不懂
仓库地址：https://github.com/Tikas/xtikas-test

**Reply by Mister-Hope** [reply]
> 哇，大佬竟然这么快回复，由于第一次使用 vuepree，也是第一次使用构建命令，确实很多不懂
> 
> 仓库地址：https://github.com/Tikas/xtikas

你这是编译后的文件啊，我要看你提交到~~vercel~~cloudframe的仓库

**Reply by Tikas** [reply]
其实，我已经删仓重建了，真的。

回复当天一直是只包含 docs 及 package.json 和 yarn.lock 文件，然而，一直失败

![1](https://user-images.githubusercontent.com/3396710/145565196-61d85db1-e6e8-4e49-a2c1-1def8c757581.jpg)

那怕是使用 yarn run build，依旧失败，能访问 docs 目录下所有生成的 html，但是，主题是官方的

vuepress build docs 会一直卡在构建应用，有次都等一小时多了，还在构建，实在是没办法了

![2](https://user-images.githubusercontent.com/3396710/145565415-8936d427-b5db-4163-815c-6dc4ff98f7b1.jpg)

**Reply by Tikas** [reply]
上面两个截图是仅有的截图了，当时也是在和别人讨论怎么解决，后来没办法了，就直接上传生成的文件了，不再折腾构建了。

**Reply by Tikas** [reply]
另外，我怀疑这是 CF 的问题，可能我太菜了，不管是怎么换构建命令，总是失败，或者成功也是官方的主题。

还有另外：本地构建是没问题的，真的，我都删除本地直接 clone 仓库的回来，使用 cmd 依旧能成功，这不是一次的试验。

我每次换一个构建命令都会进行本地尝试，本地没问题才 push 仓库，触发 cf 的构建，结果，可想而知：失败

**Reply by Mister-Hope** [reply]
从你目前的截图来看，这完全是你的build命令的问题，你应该直接试图运行 `yarn` 或者 `yarn install` 命令来完成命令的安装。另外，使用 `yarn run docs:build` 来构建文档结果。

所以请尝试重新上传你的原仓库到 CFP，并尝试替换 build 命令为 `yarn && yarn docs:build`。(注，我没用过 CFP，但我觉得它会像其他环境一样，在检测到package.json时在初始化的时候自动安装依赖，所以命令前一半应该是可选的。

**Comment by Mister-Hope** [comment, marked as answer]
问题来源于错误的构建命令

**Reply by Tikas** [reply]
大大，你好，我把仓库恢复到最初时候，依旧失败

大概配置如下：

cloudflare page：

+ 构建命令：yarn run build
+ 构建输出目录：/dist

package.json：

+ build：yarn install && yarn docs:build
+ docs:build：vuepress build docs

.vuepress/config.js

+ dest：./dist

依旧是本地 clone 回来，执行 yarn run build 构建成功

CF 会无限停留在构建应用程序中：

![Snipaste_2021-12-12_14-14-23](https://user-images.githubusercontent.com/3396710/145702490-d961972a-e922-4bde-8ed7-4cf691889f53.jpg)

仓库地址：https://github.com/Tikas/xtikas-test

**Reply by Tikas** [reply]
使用过的构建命令有：

+ vuepress build 
+ vuepress build docs
+ yarn docs:build
+ yarn run build
 - build 里尝试过如前面图所示的安装 vuepress 和 hope 去构建，当然也尝试过正常的构建命令，yarn install && yarn docs:build 是后面大大您提示后才尝试的

使用过的输出目录就比较多了，有 docs 目录下的，有 docs/.vuepress 目录下的，也有输出到根目录的，当然输出目录名统一使用：dist

这些都失败了

**Reply by Mister-Hope** [reply]
那可能是cfp的问题，因为这是个盲盒，我没有办法进行调试，所以我也帮不了你.... 我个人建议你用 Github Actions 进行构建，然后再通过git hook 触发部署。

**Reply by Tikas** [reply]
是的，我也是怀疑是 CF 问题

**Reply by uebian** [reply]
您好，我也遇到了类似的问题，麻烦问一下现在有解决方案了吗？

**Reply by Tikas** [reply]
### 写在前面

您好，我仅有一个最为基础的解决方案，希望能帮上您：

我是另外新建一个仓库，用于 push 存放静态文件，然后 CFG 直接进行部署。

由于没能去研究 @Mister-Hope  说的使用 Github Actions 构建，再 hook 到存放静态文件的仓库来触发 Cloudflare page 部署，所以在这里使用的方法是直接 Cloudflare page 直接部署静态文件。

### Cloudflare page 构建设置

构建命令：exit 0
构建输出目录：/
根目录：/
环境变量：无

### 写在后面

如果您有时间能写好 Github Actions 构建，hook 到另一个仓库（不知道 CFG 是否支持同仓的不同分支来构建）触发部署，不妨通知一声，或者这里分享一下您的自动部署方案的文件地址。

感谢感谢！

（懒已经成为我的本能....）

**Reply by uebian** [reply]
您好，我目前已经通过采用Cloudflare Workers解决了这个问题，感谢回复

**Reply by Tikas** [reply]
太强了，客气了，有时间有也去研究一下怎么实现

**Reply by cuberwr** [reply]
cf的node版本很低，环境变量里指定下版本npm run docs:build就能构建成功

**Reply by Tikas** [reply]
主要是 github 的自动部署都还不会用，我平时的时间忙于工作，这些方面，没去进一步学习，只能等以后能开发主题时，再来补上，感谢您提供的方案，后续会留意相关版本问题。
